clifford/_layout.py

```diff
@@ -157,6 +157,18 @@ def __init__(self, sig, bladeTupList, firstIdx=0, names=None):
         self.dual_func = self.gen_dual_func()
         self.vee_func = self.gen_vee_func()
 
+    @classmethod
+    def _from_sig(cls, sig=None, *, firstIdx=1, **kwargs):
+        """ Factory method that uses sorted blade tuples.  """
+        bladeTupList = cf.elements(len(sig), firstIdx)
+
+        return cls(sig, bladeTupList, firstIdx=firstIdx, **kwargs)
+
+    @classmethod
+    def _from_Cl(cls, p=0, q=0, r=0, **kwargs):
+        """ Factory method from a ${Cl}_{p,q,r}$ notation """
+        return cls._from_sig([0]*r + [+1]*p + [-1]*q, **kwargs)
```

Comment:
For another time - decide if these make better sense as a public API than the current `clifford.Cl`, and if so come up with succinct names (and remove the leading `_`

Reply:
i like them as public methods. not sure of a good name for the `from_pqr()`, but that it used most i would guess.

Reply:
Perhaps just `Layout.Cl(...)`